components/cache/adapters/redis_adapter.rst

```diff
@@ -67,12 +67,18 @@ replaced by ``rediss`` (the second ``s`` means "secure").
 
 .. note::
 
-    A `Data Source Name (DSN)`_ for this adapter must use the following format.
+    A `Data Source Name (DSN)`_ for this adapter must use either one of the following formats.
 
     .. code-block:: text
 
         redis[s]://[pass@][ip|host|socket[:port]][/db-index]
 
+    .. code-block:: text
+
+        redis[s]:[[user]:pass@]?[ip|host|socket[:port]][&params]
```

Comment:
When having this example we can drop the previous one, right?

Reply:
Well the formats differ slightly, so I don't think removing the first example is a good idea.

Example:

```
# First exmample; note the `://` and database index notation with `/1`
redis://default:password@redis:6379/1

# Second example; note the `?` and using `dbindex=1`. Basically syntax using query parameters.
redis:default:password@?host[redis:6379]&dbindex=1"
```

The first syntax is simpler, and think the majority of users who are just using a single Redis instance for basic caching, still use this format. The second syntax was added as it's derived from the format for multiple DSNs, which is a bit more complicated at first glance, but it can also be used for single Redis instance. Removing the first example would probably cause confusion again. They are both valid, just slightly different. 😄

Reply:
@xabbuh What do you think? Do you still have remarks or can we continue to complete this PR?